frontend/rust-lib/flowy-database2/src/services/group/controller.rs

```diff
@@ -396,6 +396,24 @@ where
     Ok(None)
   }
 
+  fn delete_group(&mut self, group_id: &str) -> FlowyResult<(Vec<RowId>, Option<TypeOptionData>)> {
```

Comment:
```rust
fn delete_group(&mut self, group_id: &str) -> FlowyResult<(Vec<RowId>, Option<TypeOptionData>)> {
    // Check if the group_id is not the field_id and the group exists
    let group = if group_id != self.field_id() {
        self.get_group(group_id)
    } else {
        None
    };

    match group {
        Some((_index, group_data)) => {
            let row_ids = group_data.rows.iter().map(|row| row.row.id.clone()).collect();
            let type_option_data = self.delete_group_custom(group_id)?;
            Ok((row_ids, type_option_data))
        }
        None => Ok((vec![], None)),
    }
}

```
1. reduce get_group calls
2. use of match on the group variable directly handles the cases where the group exists or not